barretenberg/bbup/bbup.ts

```diff
@@ -0,0 +1,71 @@
+#!/usr/bin/env node
+import { Command, Option } from "commander";
+const program = new Command();
+import { installBB } from "./shell.js";
+import ora from "ora";
+import logSymbols from "log-symbols";
+import { getBbVersionForNoir } from "./versions.js";
+import { execSync } from "child_process";
+
+const spinner = ora({ color: "blue", discardStdin: false });
+
+const bbup = program
+  .command("install", { isDefault: true })
+  .description("Installs Barretenberg.")
+  .addOption(
+    new Option(
+      "-v, --version <version>",
+      "The Barretenberg version to install"
+    ).implies({ noirVersion: null })
+  )
+  .addOption(
+    new Option(
+      "-nv, --noir-version <version>",
+      "The Noir version to match"
+    ).default("current")
+  )
+  .action(async ({ version, noirVersion }) => {
+    let resolvedBBVersion = "";
+    if (noirVersion) {
+      let resolvedNoirVersion = noirVersion;
+      if (noirVersion === "current") {
+        spinner.start(`Querying noir version from nargo`);
+        try {
+          const output = execSync("nargo --version", { encoding: "utf-8" });
+          resolvedNoirVersion = output.match(
+            /nargo version = (\d+\.\d+\.\d+)/
```

Comment:
This is going to break when we move to a pre-release workflow.

Reply:
Just keep me posted and I'll find a way somehow :)